apps/sms/views/conversation/js/conversation.js

```diff
@@ -547,34 +562,39 @@ var ConversationView = {
   },
 
   beforeEnterThread: function conv_beforeEnterThread(args) {
-    // TODO should we implement hooks to Navigation so that Threads could
-    // get an event whenever the panel changes?
-    Threads.currentId = args.id;
+    var threadId = +args.id;
+    return MessageManager.ensureThreadRegistered(threadId).then(() => {
```

Comment:
I see few lines below we request ```Threads.active``` at least that I suppose may be null if we opened app from notification and haven't registered Thread yet.